### PR TITLE
coccinelle: Add script to remove unnecessary return variable

### PR DESCRIPTION
**Pull request by himanshujha199640**
This script helps to remove bogus intermediate local variable
 used in functions to store return value and instead return
 directly while saving few bits of memory.

### CODE REVIEW

**Comment by pfalcon** [comment]
> If it's obvious, then you get shorter code that's just as readable (= more readable).

But less maintainable. We can't add debug logging everywhere, but adding adhoc debug here and there is a natural part of development.

**Comment by ulfalizer** [comment]
> But less maintainable. We can't add debug logging everywhere, but adding adhoc debug here and there is a natural part of development.

Only takes a few seconds to add a temporary variable if you need it. Probably won't need it there though, since it's already obvious what it returns.

I'm kinda extremist when it comes to removing as much code as possible though. I hate code. :P

**Comment by himanshujha199640** [comment]
> > while saving few bits of memory.
> 
> No memory is wasted, no worries. It's among the most trivial compiler optimizations to propagate constants, and every semi-decent compiler did it for a while.

I'm not sure to understand but you mean compiler can optimize away the local
bogus variable ?

Does `status` variable gets optimized away here:https://github.com/zephyrproject-rtos/zephyr/pull/11637/commits/218db8d92db939aa42f2b6325ec08b3d1e5c548d ?

In fact, in linux kernel we don't even allow an unnecessary initialisation `=0` to a local
variable if its not required. Its only allowed when in any code path there can be a
unitialised scalar read. 

They say initialisation to 0 also has some costs(cycles) and `-O2` `-O3` are not smart
enough to figure that out.

> > This script helps to remove bogus intermediate local variable
> > used in functions to store return value and instead return
> > directly
> 
> Can you confirm that your script does that only for constant values and nothing else? From a quick look at the Coccinelle script, that appears to be the case, but then I know too little about it (need to learn it one sweet day, yeah, when only it comes...)
> 
> Because for the code like:
> 
> ```
> int ret = some_func();
> return ret;
> ```

Ofcourse, we need such cases in conditions if `some_func()` might fail and therefore direct
return is not preferred in such a case.

But I assume you didn't look at my commits carefully. 
I don't consider such cases and remove only the unnecessary ones.

**Comment by ulfalizer** [comment]
> I'm not sure to understand but you mean compiler can optimize away the local
> bogus variable ?

Yeah, it'll get optimized away. The compiler is allowed to assume that `k_sleep()` can't touch the local `status` variable. It's still a big +1 from me though, just from a code readability perspective.

There's [an awesome site](https://godbolt.org/z/2IrR5S) for checking what the compiler spits out by the way.

File: subsys/net/l2/openthread/openthread.c [hunk 1]
```diff
@@ -232,7 +232,6 @@ static enum net_verdict openthread_recv(struct net_if *iface,
 enum net_verdict openthread_send(struct net_if *iface, struct net_pkt *pkt)
 {
 	struct openthread_context *ot_context = net_if_l2_data(iface);
-	enum net_verdict ret = NET_OK;
```
Comment (tbursztyka) [inline subsys/net/l2/openthread/openthread.c]: Change the commit title prefix to "net: openthread: "

**Review by ulfalizer — APPROVED** [review]

File: subsys/net/l2/openthread/openthread.c (hunk 1 above)
Comment (himanshujha199640) [reply subsys/net/l2/openthread/openthread.c]: Done!

**Review by jukkar — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Pull request merged**